src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -480,21 +507,37 @@ private string CreateTokenPrivate(JObject payload, SigningCredentials signingCre
                 header.Merge(JObject.FromObject(additionalHeaderClaims));
 
             var rawHeader = Base64UrlEncoder.Encode(Encoding.UTF8.GetBytes(header.ToString(Formatting.None)));
-
-            if (SetDefaultTimesOnTokenCreation)
+            try
```

Comment:
Above, CreateDefaultJWSHeader takes the 'tokenType'.
It's possible that a user wants 'typ' to be different in each header.

Reply:
Addressed it through AdditionalInnerHeaderClaims.